### PR TITLE
Alerting: Add TS types for k8s time interval API

### PR DESCRIPTION
**Pull request by tomratcliffe**
(related to https://github.com/grafana/grafana/pull/88201)

### CODE REVIEW

**Comment by tomratcliffe** [comment]
Here I tweaked Yuri's PR to set `frontend: true`, and then ran this in the root:

`grafana-app-sdk generate  -c apps/alerting/notifications -g ./apis --crdencoding none -t public/app/features/alerting/unified/api/k8s`

File: public/app/features/alerting/unified/api/k8s/timeinterval/v0alpha1/types.metadata.gen.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * metadata contains embedded CommonMetadata and can be extended with custom string fields
+ * TODO: use CommonMetadata instead of redefining here; currently needs to be defined here
+ * without external reference as using the CommonMetadata reference breaks thema codegen.
+ */
+export interface Metadata {
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/api/k8s/timeinterval/v0alpha1/types.metadata.gen.ts:6]: Where do the names and comments come from exactly?

**Comment by tomratcliffe** [comment]
(Closing, was mainly done out of interest but this won't be done straight away)

File: public/app/features/alerting/unified/api/k8s/timeinterval/v0alpha1/types.metadata.gen.ts (hunk 1 above)
Comment (yuri-tceretian) [reply public/app/features/alerting/unified/api/k8s/timeinterval/v0alpha1/types.metadata.gen.ts:6]: generator [in app-sdk](https://github.com/grafana/grafana-app-sdk) https://github.com/grafana/grafana-app-sdk/blob/9303f0a9fb2f1fbd3242e0a4b93c6d24111f5558/codegen/cuekind/def.cue#L26

**Comment by yuri-tceretian** [comment]
I think we will need it soon. When we will transfer to the new UI. Right?

**Comment by tomratcliffe** [comment]
I think there isn't a strict need for us to have this at the start, but it's useful for us to have as a reference. When I get to this work I'll perhaps revisit including the types? But ideally I might want some of them named slightly differently but I don't know for sure until I start integrating with the new API :+1:

**Comment by yuri-tceretian** [comment]
Field names were chosen to match the legacy models to a little bit simplify migration (hopefully :) )

**Comment by tomratcliffe** [comment]
Yes sorry, the field names/structure of the object matches up with the existing one nicely, but the generated types are a bit generically named in some places (e.g. `Spec` is quite loosely named)

**Comment by yuri-tceretian** [comment]
> (e.g. Spec is quite loosely named)

ah, I agree. I think they follow the way we generate backend models where we put every model in its own package, and then address them like `timeinterval.Spec`. This does not work well in FE. 
cc @IfSentient WDYT?

**Pull request closed without merging**